gax-java/gax/src/main/java/com/google/api/gax/rpc/Watchdog.java

```diff
@@ -220,6 +220,12 @@ public void onStartImpl(StreamController controller) {
             public void disableAutoInboundFlowControl() {
               Preconditions.checkState(
                   !hasStarted, "Can't disable automatic flow control after the stream has started");
+
+              // WatchdogStream is created with waiting state. If auto flow control is disabled,
+              // set the state to be IDLE, which will be updated to WAITING when onRequest() is
+              // called.
```

Comment:
I think you want to move the first part of this info to the initialization of the variable and keep the second part here:
"When a caller disables automatic flow control, they are responsible for requesting the first message. So the initial state for the watchdog is idle.